src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/ValidationQueryFilterAndParameterParserAttribute.cs

```diff
@@ -0,0 +1,149 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Health.Fhir.Api.Configs;
+using Microsoft.Health.Fhir.Api.Features.Routing;
+using Microsoft.Health.Fhir.Core.Features;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Features.Validation;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Api.Features.Filters
+{
+    [AttributeUsage(AttributeTargets.Class)]
+    internal class ValidationQueryFilterAndParameterParserAttribute : ActionFilterAttribute
+    {
+        private readonly FeatureConfiguration _features;
+
+        public ValidationQueryFilterAndParameterParserAttribute(IOptions<FeatureConfiguration> features)
+        {
+            EnsureArg.IsNotNull(features, nameof(features));
```

Comment:
nit: both validations can be done in one call by checking for `features?.Value`

Reply:
Changed